lib/xandra/connection.ex

```diff
@@ -190,9 +193,15 @@ defmodule Xandra.Connection do
 
     with :ok <- state.transport.send(socket, payload),
          {:ok, %Frame{} = frame} <-
-           Utils.recv_frame(state.transport, socket, state.protocol_module, compressor) do
-      {:ok, query, %{frame | atom_keys?: atom_keys?}, state}
+           Utils.recv_frame(state.transport, socket, state.protocol_module, compressor),
+         frame = %{frame | atom_keys?: atom_keys?},
+         %SetKeyspace{keyspace: keyspace} = response <-
+           state.protocol_module.decode_response(frame, query, options) do
+      {:ok, query, response, %{state | current_keyspace: keyspace}}
     else
+      %_{} = response ->
```

Comment:
@whatyouhide here I was not sure if I have to explicitly list all the possible structs that can be returned by `decode_response`, like
```elixir
%kind{} when kind in [...]
```
let me know if you prefer it that way or, option 3, if I should skip the last line in the `with` statement and do something like
```elixir
case state.protocol_module.decode_response(frame, query, options) do
  %SetKeyspace{keyspace: keyspace} ->
    {:ok, query, response, %{state | current_keyspace: keyspace}}

  response ->
    {:ok, query, response, state}
end
```